src/ServiceControl.Monitoring/Http/HttpEndpoint.cs

```diff
@@ -52,9 +52,13 @@ class NancyTask : FeatureStartupTask
 
             public NancyTask(IBuilder builder, Uri host)
             {
-                var buildstrapper = new Bootstrapper(builder.BuildAll<IEndpointDataProvider>());
+                var bootstrapper = new Bootstrapper(
+                    builder.Build<QueueLengthDataStore>(), 
```

Comment:
I've removed the dependency from `IBuilder` and now we are passing actions that do concrete type registrations.